Resources/doc/sync/descriptors/descriptors.rst

```diff
@@ -0,0 +1,122 @@
+
+    # Relation collection service must be configured in order to use it.
+    ongr_connections.sync.extraction_collection:
+        class: ONGR\ConnectionsBundle\Sync\Extractor\Descriptor\ExtractionCollection
+
+
+Example simple extraction Descriptors setup:
+
+.. code-block:: yaml
+
+    services:
+        # Descriptor for category creation.
+        my_project.extractor.descriptors.category.create:
+            class: %ongr_connections.extractor.descriptor.class%
+            arguments: [my_categories, C, 1, category, NEW.categories_id]
+            tags:
+                - { name: ongr_connections.extraction_descriptor }  # This tag is used to collect all descriptors.
+
+        # Descriptor for category update.
+        my_project.extractor.descriptors.category.update:
+            class: %ongr_connections.extractor.descriptor.class%
+            arguments: [my_categories, U, 1, category, NEW.categories_id]
+            tags:
+                - { name: ongr_connections.extraction_descriptor }  # This tag is used to collect all descriptors.
+
+        # Descriptor for category deletion
+        my_project.extractor.descriptors.category.delete:
+            class: %ongr_connections.extractor.descriptor.class%
+            arguments: [my_categories, D, 1, category, OLD.categories_id]
+            tags:
+                - { name: ongr_connections.extraction_descriptor }  # This tag is used to collect all descriptors.
+
+..
+
+Extraction Descriptor constructor arguments are as follows:
+
+.. code-block:: php
+
+    /**
+     * @param string      $table        Table name to hook on.
+     * @param string      $type         Trigger and default job type C - create, U - update,  D - delete.
+     * @param int|null    $idField      Source for document id.
+     * @param int|null    $updateType   Partial update - 0, full update - 1.
+     * @param string|null $documentType Type of target document.
+     * @param array       $trackFields  Array of table fields to track, all using default priority.
+     * @param string|null $jobType      C - create, U - update,  D - delete.
+     */
+..
+
+Last two parameters can and should be left undefined, as they are to be deprecated in near future.
+
+Cascading changes
+-----------------
+
+There are classes that help implement cascading data changes, e.g. if you have changed the name of the
+category, you also have to update all the products which use said categories' name.
+
+``Extractor\Descriptor\JoinRelation`` class does exactly that: you can attach it to a descriptor and when changes are
+detected, `extractor <../extractor/extractor.rst>`_ calls the relations and marks related documents as changed.
+
+If ``$documentType`` is left undefined document itself will not be marked as changed but related items will.
+
+Example cascading change configuration:
+
+.. code-block:: yaml
+
+    parameters:
+        ongr_connections.extractor.join_relation.class: ONGR\ConnectionsBundle\Sync\Extractor\Relation\JoinRelation
```

Comment:
this is default service class parameter, we shouldn't add it in example, to not to confuse users.